packages/odyssey-react-mui/src/Button.tsx

```diff
@@ -154,6 +156,12 @@ const Button = ({
 }: ButtonProps) => {
   const muiProps = useMuiProps();
   const localButtonRef = useRef<HTMLButtonElement>(null);
+  const buttonContext = useButton();
+  const isFullWidth = useMemo(
+    () =>
+      buttonContext.isFullWidth ? buttonContext.isFullWidth : isFullWidthProp,
+    [buttonContext, isFullWidthProp]
+  );
```

Comment:
This is kinda funny of me to say, but this doesn't need to be a `useMemo` because it's a primitive. Typically, you only wanna memoize high-compute task (like array and object manipulations) or when the resulting value is an object or array.

Primitives don't change memory references; thus, won't trigger a re-render unless the value changes.

Although, sometimes, having these memoized in the first place just means that it's harder to mess them up in the future. No one can come in here, change this boolean to an object, and have it break for instance.

But in this case, I think we're safe just doing that logic inline in the component.